src/components/common/UserProfile/UserProfile.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import ProfileTemplate from '~frame/ProfileTemplate';
+import NotFoundRoute from '~routes/NotFoundRoute';
+import { useUserByNameOrAddress } from '~hooks';
+
+import UserMeta from './UserMeta';
+import UserProfileSpinner from './UserProfileSpinner';
+import UserColonies from './UserColonies';
+
+import styles from './UserProfile.css';
+
+const displayName = 'common.UserProfile';
+
+const UserProfile = () => {
+  const { username } = useParams();
+  const { user, loading, error } = useUserByNameOrAddress(username || '');
+
+  if (loading) {
+    return <UserProfileSpinner />;
+  }
+
+  if (error || !user) {
+    return <NotFoundRoute />;
+  }
```

Comment:
Until you address the rest of the changes, I'll test this some more locally, as to maybe find some solution to the "redirect to landing page" issue